File: ValidPalindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        String res = "";
        for(char c : s.toCharArray()){
            if(Character.isDigit(c) || Character.isLetter(c)){
                res = res + c;
            }
        }
        
        res = res.toLowerCase();
        
        int leftPointer = 0;
        int rightPointer = res.length()-1;
        
        while(leftPointer <= rightPointer){
            if(res.charAt(leftPointer) != res.charAt(rightPointer)){
                return false;
            }
            
            leftPointer++;
            rightPointer--;
            
        }
        
        return true;
        
    }
}

File: ValidPalindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
 // TC: O(N) SC: O(1)
var isPalindrome = function(s) {
    const isLetter = (ch) =>  (ch.charCodeAt(0) >= 97 && ch.charCodeAt(0) <= 122) ? true : false;
    const isDigit = (ch) =>  (ch.charCodeAt(0) >= 48 && ch.charCodeAt(0) <= 57) ? true : false;
    
    s = s.toLowerCase();
    let start = 0, end = s.length - 1;

    while(start < end) {
        if((isLetter(s[start]) || isDigit(s[start])) && (isLetter(s[end]) || isDigit(s[end]))) {
            if(s[start] !== s[end]) {
                return false;
            }
            else {
                start++;
                end--;
            }
        }
        else if(!(isLetter(s[start]) || isDigit(s[start]))) {
            start++;
        }
        else if(!(isLetter(s[end]) || isDigit(s[end]))) {
            end--;
        }
    }
    return true;
};
